core/services/job/common.go

```diff
@@ -0,0 +1,33 @@
+package job
+
+import (
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+)
+
+//go:generate mockery --name Spec --output ./mocks/ --case=underscore
+//go:generate mockery --name Service --output ./mocks/ --case=underscore
+
+type (
+	Type string
+
+	Spec interface {
```

Comment:
Subsequent PR, because it has become clear that we need to have a broader conversation on this with more stakeholders: https://www.pivotaltracker.com/story/show/175300566